imports/plugins/core/components/lib/hoc.js

```diff
@@ -40,14 +40,17 @@ export function withCurrentAccount(component) {
       return null;
     }
 
-    // shoppers should always be guests
-    const isGuest = Roles.userIsInRole(user, "guest", shopId);
-    // but if a user has never logged in then they are anonymous
-    const isAnonymous = Roles.userIsInRole(user, "anonymous", shopId);
+    const accSub = Meteor.subscribe("Accounts", user._id);
+    if (accSub.ready()) {
+      // shoppers should always be guests
+      const isGuest = Roles.userIsInRole(user, "guest", shopId);
+      // but if a user has never logged in then they are anonymous
+      const isAnonymous = Roles.userIsInRole(user, "anonymous", shopId);
```

Comment:
Should be using `Reaction.hasPermissions`

Reply:
Yep @impactmass =, just knock those two role check instances out and we can merge this.

Reply:
@jshimko I've updated the use Roles.userIsInRole for the guest check. At the moment, we can not replace the anonymous check with hasPermission. I added comments describing why. That will be worked in #2895.

Reply:
Great, thanks!